impl/client/src/main/javascript/web/prompting/components/StaticAutocompleteBoxComponent.js

```diff
@@ -118,15 +118,20 @@ define([ 'common-ui/util/util', 'cdf/components/BaseComponent',  'amd!cdf/lib/jq
           if (myself.prevSelValue !== request.term) {
             myself.prevSelValue = request.term;
             myself.dashboard.processChange(myself.name);
+            this.pendingChanges = true;
+            // Return a empty array to resolve callback
+            response([]);
+          } else {
+            var term = request.term.toUpperCase();
+            var matches = $.map(this.valuesArray, function(tag) {
+              // we need unescape label before matching (fix for special characters like as &#39; &amp; &lt; &gt)
+              if (myself._createLabelTag(tag.label).text().toUpperCase().indexOf(term) >= 0) { // PRD-3745
+                return tag;
+              }
+            });
+            response(matches);
```

Comment:
Why are you now processing anything when the new value input by the user is the same as the previous value introduced by the user? You will only have the correct valuesArray loaded if it is already the result of calling the service the last time. But the call may still be on the way...